resilience4j-ratelimiter/src/main/java/io/github/resilience4j/ratelimiter/internal/SemaphoreBasedRateLimiter.java

```diff
@@ -144,13 +145,35 @@ public boolean acquirePermission() {
     }
 
     /**
-     * {@inheritDoc}
-     * SemaphoreBasedRateLimiter is totally blocking by it's nature. So this non-blocking API isn't supported.
-     * It will return negative numbers all the time.
+     * Acquiring permissions of a specified weight is not supported in the
+     * spemaphore based implementation. Use {@link #acquirePermission()}
+     *
+     * @throws UnsupportedOperationException always for this implementation
+     */
+    @Override
+    public boolean acquirePermission(int weight) {
+        throw new UnsupportedOperationException("Acquiring permissions of a specified weight is not supported in the spemaphore based implementation");
+    }
+
+    /**
+     * Reserving permissions is not supported in the spemaphore based
+     * implementation. Semaphores are totally blocking by it's nature. So this
+     * non-blocking API isn't supported. Use {@link #acquirePermission()}
+     *
+     * @throws UnsupportedOperationException always for this implementation
      */
     @Override
     public long reservePermission() {
-        return -1;
```

Comment:
PS. I changed this behaviour to `throw new UnsupportedOperationException` because I think its more developer friendly. No one should call `reservePermissions` in a `SemaphoreBasedRateLimiter`. If some one is doing it he is not aware it is not supported and returning -1 will not explain that to him at first glance.